Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -82,6 +81,8 @@ namespace Terrain
                     {
                         PrepareMaterialData();
                     }
+
+                    AZ::Data::AssetBus::Handler::BusConnect(assetData->GetId());
```

Comment:
Add a comment to explain it is important that connecting to the AssetBus happens _after_ creating the MaterialInstance, otherwise material hot-reload could fail. We need Material's asset callbacks to happen first to reinitialize the material data, and _then_ have the TerrainFeatureProcessor get notified and use the latest material data.

(This is all assuming that BusConnect order determines callback execution order, which I'm not sure is the case).